src/mlpack/tests/recurrent_network_test.cpp

```diff
@@ -581,6 +653,125 @@ void ReberGrammarTestNetwork(const size_t hiddenSize = 4,
   BOOST_REQUIRE_GE(successes, 1);
 }
 
+/**
+ * Train the specified BRNN network and the construct a Reber grammar dataset.
+ */
+template<typename RecurrentLayerType>
+void ReberGrammarTestNetworkBRNN(const size_t hiddenSize = 4,
```

Comment:
Do you think we could pass a model instantiation to make the test code more general? In this case we could we don't have to write a special version for each network. Let me know what you think.

Reply:
Yes that does reduce quite a bit code duplication. Although I had to add one if condition as well since the layers in the model were different.